common/djangoapps/student/models.py

```diff
@@ -332,6 +337,57 @@ def user_profile_pre_save_callback(sender, **kwargs):
         user_profile.profile_image_uploaded_at = None
 
 
+@receiver(post_save, sender=UserProfile)
+def user_profile_post_save_callback(sender, **kwargs):
+    """
+    Emit analytics events after saving the
+    """
+    user_profile = kwargs['instance']
+    # pylint: disable=protected-access
+    emit_field_changed_events(user_profile, user_profile.user, sender._meta.db_table)
+
+
+def emit_field_changed_events(instance, user, db_table):
+    """ For the given model instance, save the fields that have changed since the last save.
+
+    Requires that the Model uses 'model_utils.FieldTracker' and has assigned it to 'tracker'.
+
+    Args:
+        instance (Model instance): the model instance that is being saved
+        user (User): the user that this instance is associated with
+        db_table (str): the name of the table that we're modifying
+
+    Returns:
+        None
+    """
+    excluded_fields = ['meta', 'password']
+    changed_fields = instance.tracker.changed()
+    changed_settings = {}
+    for field in changed_fields:
+        if field not in excluded_fields:
+            field_dict = {
+                field: {
+                    'old_value': unicode(changed_fields[field]),
+                    'new_value': unicode(getattr(instance, field)),
+                }
+            }
+            changed_settings.update(field_dict)
+
+    # only emit events when something has changed
+    if len(changed_settings) > 0:
+        context = {'user_id': user.id}
```

Comment:
I'm not sure how to do this at the level of my other unit tests (directly changing fields on `UserProfileFactory`s and then `.save()`ing). Is the concept of the requesting user introduced at the view layer?